implementation/src/main/java/io/smallrye/jwt/build/impl/JwtEncryptionImpl.java

```diff
@@ -64,15 +64,15 @@ public String encrypt(String keyLocation) throws JwtEncryptionException {
         } catch (Exception ex) {
             throw new JwtEncryptionException(ex);
         }
-        return key instanceof PublicKey ? encryptInternal((PublicKey) key) : encryptInternal((SecretKey) key);
+        return encryptInternal(key);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public String encrypt() throws JwtSignatureException {
```

Comment:
Ideally this should be a JWTEncryptionException, but that means the interface will have to change. I am not sure if this is a breaking change

Reply:
@bobbyphilip It is just a bug, thanks for spotting it, please update the interface, it is still a work in progress, we are tuning and optimizing things, some refactorings and other ifxes can go freely when needed